### PR TITLE
User friendly version (EXPOSUREAPP-5641)

### LINKED ISSUE #2564: Display Google Exposure Notifications System version more user-friendly
## Avoid duplicates
* [X] This enhancement request has not already been raised before
* [X] Enhancement request is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)
* [ ] If you are proposing a new feature, please do so in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)

## Current Implementation

Last reviewed for relevance on App version: 2.22.1

The CWA Android App Information provides a version number of the Google Exposure Notifications System at the bottom of the screen. Use the three-dot menu on the CWA Start Screen to access App Information.

App Information EN ENF Version

## Disadvantages of current implementation

1. The version number in the app (e.g. from the above screenshot 18210613000) does not correspond to the version number described in the [Exposure Notification release notes](https://developers.google.com/android/exposure-notifications/release-notes) e.g. v1.8.
2. To explain how to parse a full version number a detailed FAQ article https://www.coronawarn.app/en/faq/#ENF_version was published through PR #https://github.com/corona-warn-app/cwa-website/pull/207. This is more complicated than ought to be necessary.
3. The App Information display uses the abbreviation "ENF", however only Apple refers to an Exposure Notification framework (ENF). Google consistently refers to their module as "Exposure Notifications System", for example in the help directly linked to via Android Settings > Google > COVID-19 Exposure Notifications > Learn more. Google only rarely abbreviates this term and then they use "ENS" as in https://support.google.com/android/answer/9930236?hl=en 
"Google and Apple have built in safeguards to ensure that government contact tracing apps built with **ENS** cannot infer your location." or in the original program announcement https://blog.google/inside-google/company-announcements/update-exposure-notifications/.

## Suggested Enhancement

1. When an 11-digit ENS version number is retrieved from gms, parse and display the first two digits as the version number in addition to displaying the full version number in brackets.
2. Change the label from ENF to ENS.

So for instance instead of displaying:

`ENF Version: 18210613000`

that could become:

`ENS Version: v1.8 (18210613000)`

## Expected Benefits

1. It will be easier to recognize the version number of the Google Exposure Notifications System without a user having to understand the coding of VVYYWWnnnnn where 
VV = version
YY = year
WW = week
nnnnn
2. The version number displayed by the app will be directly relatable to the version number described in the [Exposure Notification release notes](https://developers.google.com/android/exposure-notifications/release-notes).
3. The FAQ https://www.coronawarn.app/en/faq/#ENF_version complexity can be reduced because it will no longer be necessary to describe to users how to decode a Google Exposure Notifications System version number by hand.

## Additional information

I made the equivalent enhancement request to Google for their UI and they have implemented it in the v1.8 (211213000) release.

---
Internal Tracking ID: [EXPOSUREAPP-5641](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5641)

### PR DESCRIPTION
**Pull request by mtwalli**
fixes #2564

## Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5641

### CODE REVIEW

**Comment by Ein-Tim** [comment]
@mtwalli 

Will you also change "ENF" to "ENS" as suggested by @MikeMcC399 in the corresponding issue?

**Comment by MikeMcC399** [comment]
@mtwalli 

The comment in the enhancement request issue https://github.com/corona-warn-app/cwa-app-android/issues/2564 is still valid:

"The App Information display uses the abbreviation "ENF", however only Apple refers to an Exposure Notification framework (ENF). Google consistently refers to their module as "Exposure Notifications System", for example in the help directly linked to via Android Settings > Google > COVID-19 Exposure Notifications > Learn more. Google only rarely abbreviates this term and then they use "ENS" as in https://support.google.com/android/answer/9930236?hl=en 
"Google and Apple have built in safeguards to ensure that government contact tracing apps built with **ENS** cannot infer your location." or in the original program announcement https://blog.google/inside-google/company-announcements/update-exposure-notifications/."

**Comment by mtwalli** [comment]
> @mtwalli
> 
> Will you also change "ENF" to "ENS" as suggested by @MikeMcC399 in the corresponding issue?

what is ENS stands for ? I see many places in the app where it  is referred to as ENF not ENS see `ENFClient` and `ENFVersion`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/information/InformationFragmentViewModel.kt [hunk 1]
```diff
@@ -35,8 +35,15 @@ class InformationFragmentViewModel @AssistedInject constructor(
...
-        val enfVersion = enfClient.getENFClientVersion()
-            ?.let { "ENF ${context.getString(R.string.information_version).format(it)}" }
+        val enfVersion = enfClient.getENFClientVersion()?.let { v ->
+            val vString = v.toString()
+            val version = "v%s.%s (%s)".format(
+                vString.getOrElse(0) { '0' },
+                vString.getOrElse(1) { '0' },
+                vString
+            )
+            "ENF ${context.getString(R.string.information_version).format(version)}"
```
Comment (MikeMcC399) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/information/InformationFragmentViewModel.kt:45]: Suggest to change "ENF" to "ENS" to conform to Google's abbreviation of the term, e.g. in https://blog.google/inside-google/company-announcements/update-exposure-notifications/.

**Review by MikeMcC399 — COMMENTED** [review]
Commented about possibility to change "ENF" to "ENS".

**Comment by mtwalli** [comment]
> @mtwalli
> 
> The comment in the enhancement request issue #2564 is still valid:
> 
> "The App Information display uses the abbreviation "ENF", however only Apple refers to an Exposure Notification framework (ENF). Google consistently refers to their module as "Exposure Notifications System", for example in the help directly linked to via Android Settings > Google > COVID-19 Exposure Notifications > Learn more. Google only rarely abbreviates this term and then they use "ENS" as in https://support.google.com/android/answer/9930236?hl=en "Google and Apple have built in safeguards to ensure that government contact tracing apps built with **ENS** cannot infer your location." or in the original program announcement https://blog.google/inside-google/company-announcements/update-exposure-notifications/."

Ah Ok, for consistency reasons and because it is historically was referred to as ENF and to avoid confusion why it is displayed as ENS while in the code it is ENF I would keep it as ENF

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/information/InformationFragmentViewModel.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/information/InformationFragmentViewModel.kt:45]: See https://github.com/corona-warn-app/cwa-app-android/pull/5556#issuecomment-1241773278

**Comment by MikeMcC399** [comment]
@mtwalli 

> what is ENS stands for ? 

ENS stands for "Exposure Notifications System" according to Google in their help text https://support.google.com/android/answer/9930236?hl=en

> I see many places in the app where it is referred to as ENF not ENS see `ENFClient` and `ENFVersion`

It seems the string building in this PR is in fact the only place where the abbreviation is exposed to the user.

I don't suggest changing the abbreviation in places where it is used in the code which are not displayed to the user, like for instance the naming of the module [coronawarnapp/nearby/ENFModule.kt](https://github.com/corona-warn-app/cwa-app-android/blob/release/2.27.x/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/ENFModule.kt).

**Comment by mtwalli** [comment]
> @mtwalli
> 
> > what is ENS stands for ?
> 
> ENS stands for "Exposure Notifications System" according to Google in their help text https://support.google.com/android/answer/9930236?hl=en
> 
> > I see many places in the app where it is referred to as ENF not ENS see `ENFClient` and `ENFVersion`
> 
> It seems the string building in this PR is in fact the only place where the abbreviation is exposed to the user.
> 
> I don't suggest changing the abbreviation in places where it is used in the code which are not displayed to the user, like for instance the naming of the module [coronawarnapp/nearby/ENFModule.kt](https://github.com/corona-warn-app/cwa-app-android/blob/release/2.27.x/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/ENFModule.kt).

Changed to `ENS` Added a comment for DEVs to avoid confusion

**Comment by MikeMcC399** [comment]
@mtwalli

> Changed to `ENS` Added a comment for DEVs to avoid confusion

Good decision! Google (ENS) and Apple (ENF) created the problem when they defined the name of their APIs slightly differently from each other.

**Review by marcauberer — APPROVED** [review]
lgtm 👍

**Pull request merged**